### PR TITLE
Layouts/ bottom nav proof of concept

### PR DESCRIPTION
**Pull request by solderq35**
I dunno if this is redundant at this point. useSegments is kind of not-smooth with the animations but yeah.

short vid demo: https://i.imgur.com/LFJnKkP.mp4

### CODE REVIEW

**Comment by solderq35** [comment]
The most notable changes are with the `_layout.js` files. Any other file changes are just for illustration / demo / testing purposes and can be removed if needed.

I did do this in `app/index.js` cause I was getting some import errors
```js
// import Leaderboard from './drills/leaderboard'
// import Description from './drills/description'
// import Stat from './drills/stat'
```

**Comment by FrankreedX** [comment]
I like the nesting one more directory idea and I think we should move forward with this.
I really like this branch and at this point, there's only some house cleaning stuff to do before this can be merged into layout.
I changed `index.js` to 
`import { Redirect } from "expo-router";

const Index = () => {
    return <Redirect href="/drill" />;
};
export default Index;`
So that the drill page is always default. It seems expo-router currently can't handle default paths very well so this will do for now.
After that, rebase onto `layouts` and this branch looks good to me!
I'd suggest making this not a draft and invite everyone to take a look at it

**Comment by solderq35** [comment]
I added the redirect on `app/index.js` and also added the `app/drill/[id]/(segments)` directory. I also moved the navigation layout logic to `app/(content)/_layout.js` as that seemed a better way to exclude the `app/(auth)` and `app/index.js` stuff from the navbar. I did keep `app/_layout.js` , editing it to just have a `<Slot>`. Because for some reason without doing this, nesting the nav layout into `(content)` added some weird padding on the bottom.

As far as redirects, in the future after implementing signin, we could also borrow from the old app's logic for redirecting users to signin if not logged in - https://github.com/Golf-Drill-Challenge-App/Golf-App/blob/archived_app/context/Auth.tsx

File: app/(content)/drill/[id]/(segments)/index.js [hunk 1]
```diff
@@ -32,7 +32,7 @@ export default function Index() {
 
     return (
         <PaperProvider>
-            <Appbar.Header statusBarHeight={0}>
+            <Appbar.Header statusBarHeight={30}>
```
Comment (solderq35) [inline app/(content)/drill/[id]/(segments)/index.js:35]: I changed this cause on iOS at least for me this button was way too close to the top of the screen and almost unclickable.

File: app/(content)/_layout.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
+import { Tabs } from "expo-router/tabs";
+import { useSegments } from "expo-router";
```
Comment (FrankreedX) [inline app/(content)/_layout.js:2]: remove this

File: app/(content)/_layout.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+          href: null,
+          title: "Drill submission directory",
+          headerShown: false,
+          tabBarStyle: {
+            display: "none",
+          },
+        }}
+      />
+
+      <Tabs.Screen
```
Comment (FrankreedX) [inline app/(content)/_layout.js:43]: I made an oversight of this in the figma. Can the user result have the tabbar?

**Comment by FrankreedX** [comment]
I think I've found a potential problem. When you're on the Drill tab and then click on a drill, the "Drill" button on the navbar is not highlighted anymore
Screenshot 2024-02-12 at 7 05 53 PM

File: app/(content)/_layout.js (hunk 3 above)
Comment (solderq35) [reply app/(content)/_layout.js:22]: Did you mean `app\(content)\drill\[id]\users\[user].js`? I put tabbar back for that one. 

If you meant `app\(content)\drill\[id]\submission\result.js`, I can switch later

File: app/(content)/_layout.js (hunk 2 above)
Comment (solderq35) [reply app/(content)/_layout.js]: removed the useSegments import if that's what you meant

**Comment by solderq35** [comment]
> I think I've found a potential problem. When you're on the Drill tab and then click on a drill, the "Drill" button on the navbar is not highlighted anymore

I think I was able to fix this issue, and also add the stack animations, by following this post: https://stackoverflow.com/a/77251731
Although this is just for stack animations between pages on the same tab, I don't think it is possible to have stack animations between different tabs

**Pull request closed without merging**